src/chips/nrf51/src/clock.rs

```diff
@@ -0,0 +1,180 @@
+//! The clock peripheral of the nRF51 series (chapter 13 of
+//! the nRF51 reference manual v3.0), which manages the 
+//! low frequency and high frequency clocks. The low frequency
+//! clock drives the real time clock (RTC), while the 
+//! high frequency clocks drive the timer system.
+//!
+//! Author: Philip Levis
+//! Date: August 18, 2016
+
+use common::VolatileCell;
+use common::take_cell::TakeCell;
+use core::mem;
+
+struct Registers {
+    pub tasks_hfclkstart    : VolatileCell<u32>,
+    pub tasks_hfclkstop     : VolatileCell<u32>,
+    pub tasks_lfclkstart    : VolatileCell<u32>,
+    pub tasks_lfclkstop     : VolatileCell<u32>,
+    pub tasks_cal           : VolatileCell<u32>,
+    pub tasks_cstart        : VolatileCell<u32>,
+    pub tasks_cstop         : VolatileCell<u32>,
+    _reserved1: [VolatileCell<u32>; 57],
+    pub events_hfclkstarted : VolatileCell<u32>,
+    pub events_lfclkstarted : VolatileCell<u32>,
+    pub done                : VolatileCell<u32>,
+    pub ctto                : VolatileCell<u32>,
+    _reserved2: [VolatileCell<u32>; 125],
+    pub intenset            : VolatileCell<u32>,
+    pub intenclr            : VolatileCell<u32>,
+    _reserved3: [VolatileCell<u32>; 63],
+    pub hfclkrun            : VolatileCell<u32>,
+    pub hfclkstat           : VolatileCell<u32>,
+    _reserved4: [VolatileCell<u32>; 1],
+    pub lfclkrun            : VolatileCell<u32>,
+    pub lfclkstat           : VolatileCell<u32>,
+    pub lfclksrccopy        : VolatileCell<u32>,
+    _reserved5: [VolatileCell<u32>; 62],
+    pub lfclksrc            : VolatileCell<u32>,
+    _reserved6: [VolatileCell<u32>; 7],
+    pub ctiv                : VolatileCell<u32>,
+    _reserved7: [VolatileCell<u32>; 5],
+    pub xtalfreq            : VolatileCell<u32>,
+}
+
+const CLOCK_BASE: usize = 0x40000000;
+
+pub enum InterruptField {
+    HFCLKSTARTED = (1 << 0),
+    LFCLKSTARTED = (1 << 1),
+    DONE         = (1 << 3),
+    CTTO         = (1 << 4),
+}
+
+pub enum ClockTaskTriggered {
+    NO   = 0,
+    YES  = 1,
+}
+
+pub enum ClockRunning {
+    NORUN   = 0,
+    RUN     = (1 << 16),
+}
+
+pub enum LowClockSource {
+    RC            = 0,
+    XTAL          = 1,
+    SYNTH         = 2,
+    MASK          = 0x3,
+}
+
+pub enum HighClockSource {
+    RC            = 0,
+    XTAL          = 1,
+}
+
+pub enum XtalFreq {
+    F16MHz         = 0xFF,
+    F32MHz         = 0,
+}
+
+pub struct Clock {
+    client: TakeCell<&'static ClockClient>,
+}
+
+pub trait ClockClient {
+    /// All clock interrupts are control signals, e.g., when
+    /// a clock has started etc. We don't actually handle any
+    /// of them for now, but keep this trait in place for if we
+    /// do need to in the future.
+    fn event(&self);
+}
+
+pub static mut CLOCK : Clock = Clock {
+    client: TakeCell::empty(),
+};
+
+
+impl Clock {
+    #[allow(non_snake_case)]
+    fn CLOCK(&self) -> &'static Registers {
```

Comment:
perhaps a better name would be `regs()` or `registers()`? Alternatively, you could also adopt @lizardo's style of having a global `CLOCK()` function rather than a method. Finally, @KBaichoo and I have been experimenting with a special `StaticRef` type (see an example [here](https://github.com/alevy/tock-hotel/blob/driver/usb/src/chips/hotel/src/usb/mod.rs#L11)) but I don't suggesting using it yet as it's still pretty experimental.